### PR TITLE
msbuild patch to remove BuildXL feed

### PR DESCRIPTION
**Pull request by mthalman**
Backport of https://github.com/dotnet/sdk/pull/44194

### CODE REVIEW

File: src/SourceBuild/patches/msbuild/0001-Remove-BuildXL-feed.patch [hunk 1]
```diff
@@ -0,0 +1,22 @@
+From 6e9ddc58ab1d805c97e38be2c1978c927ba4bd64 Mon Sep 17 00:00:00 2001
+From: Matt Thalman <mthalman@microsoft.com>
+Date: Tue, 15 Oct 2024 14:24:55 -0500
+Subject: [PATCH] Remove BuildXL feed
+
+Backport: https://github.com/dotnet/msbuild/pull/10765
```
Comment (ViktorHofer) [inline src/SourceBuild/patches/msbuild/0001-Remove-BuildXL-feed.patch:6]: This links to the main PR. I see that the line still exists in vs17.12: https://github.com/dotnet/msbuild/blob/0258a80c00f599c2b0564d0429cd52f4d815e83c/NuGet.config#L21

cc @JanKrivanek @rainersigwald

Comment (mthalman) [reply src/SourceBuild/patches/msbuild/0001-Remove-BuildXL-feed.patch:6]: Yes, I've already responded at https://github.com/dotnet/msbuild/pull/10765#issuecomment-2414826373 that this needs to be backported.

Comment (JanKrivanek) [reply src/SourceBuild/patches/msbuild/0001-Remove-BuildXL-feed.patch:6]: Do we source build only 1xx versions? Is 8.0.1xx enough (so backporting to 17.8 https://learn.microsoft.com/en-us/dotnet/core/porting/versioning-sdk-msbuild-vs#versioning)? Or do we need it in all 8.0 versions?

Comment (mthalman) [reply src/SourceBuild/patches/msbuild/0001-Remove-BuildXL-feed.patch:6]: Just the 1xx versions should be sufficient.

Comment (JanKrivanek) [reply src/SourceBuild/patches/msbuild/0001-Remove-BuildXL-feed.patch:6]: FYI @dotnet/kitten

**Review by ellahathaway — APPROVED** [review]

**Pull request merged**